external/loaders/loaders/mappers/_fine_res.py

```diff
@@ -0,0 +1,236 @@
+        merged["Q1"] = _extend_lower(merged["Q1"])
+        merged["Q2"] = _extend_lower(merged["Q2"])
+
+    return _ml_standard_names(merged)
+
+
+def _add_nudging_tendencies(merged: xr.Dataset):
+    with xr.set_options(keep_attrs=True):
+        Q1 = merged.Q1 + merged.air_temperature_tendency_due_to_nudging
+        Q2 = merged.Q2 + merged.specific_humidity_tendency_due_to_nudging
+    Q1.attrs.update(
+        {
+            "long_name": merged.Q1.attrs.get("long_name")
+            + " plus dynamics nudging tendencies",
+            "description": merged.Q1.attrs.get("description")
+            + " + dynamics nudging tendencies",
+        }
+    )
+    Q2.attrs.update(
+        {
+            "long_name": merged.Q2.attrs.get("long_name")
+            + " plus dynamics nudging tendencies",
+            "description": merged.Q2.attrs.get("description")
+            + " + dynamics nudging tendencies",
+        }
+    )
+    return Q1, Q2
+
+
+def _add_dynamics_differences(merged: xr.Dataset):
+    with xr.set_options(keep_attrs=True):
+        Q1 = merged.Q1 + merged.fine_minus_coarse_tendency_of_air_temperature_due_to_dynamics
+        Q2 = merged.Q2 + merged.fine_minus_coarse_tendency_of_specific_humidity_due_to_dynamics
+    Q1.attrs.update(
+        {
+            "long_name": merged.Q1.attrs.get("long_name")
+            + " plus dynamics differences",
+            "description": merged.Q1.attrs.get("description")
+            + " + dynamics differences",
+        }
+    )
+    Q2.attrs.update(
+        {
+            "long_name": merged.Q2.attrs.get("long_name")
+            + " plus dynamics differences",
+            "description": merged.Q2.attrs.get("description")
+            + " + dynamics differences",
+        }
+    )
+    return Q1, Q2
+
+
+def _extend_lower(
+    fine_source: xr.DataArray, vertical_dim: str = "z"
+) -> xr.Dataset:
+    if fine_source.sizes[vertical_dim] < 2:
+        raise ValueError("vertical_dim must be greater than 1.")
+    fine_source_new_bottom = fine_source.isel({vertical_dim: -2})
+    fine_source_without_bottom = fine_source.isel({vertical_dim: slice(None, -1)})
+    fine_source_extended_lower = xr.concat(
+        [fine_source_without_bottom, fine_source_new_bottom], dim=vertical_dim
+    )
+    fine_source_extended_lower.attrs.update(
+        {
+            "long_name": fine_source.attrs.get("long_name")
+            + " with lowest layer overriden",
+            "description": fine_source.attrs.get("description")
+            + ", with lowest layer overriden",
+        }
+    )
+    return fine_source_extended_lower
+
+
+def _ml_standard_names(merged: xr.Dataset):
+
+    # since ML target is Q1/Q2, dQ1=Q1 and pQ1=0 and same for moistening
+    merged["dQ1"] = merged["Q1"]
+    merged["dQ2"] = merged["Q2"]
+    merged["pQ1"] = xr.zeros_like(merged.Q1)
+    merged["pQ1"].attrs = {"units": "K/s", "long_name": "coarse-res physics heating"}
```

Comment:
This is copied over but wondering if we can drop the pQs now that they're in the derived mapping?

Reply:
These renames were done for the offline diags, but I don't have a great sense of whether they are actually necessary. Is just leaving ML target as Q1/Q2 and not having a dQ1/dQ2 okay? @AnnaKwa

Reply:
My reading of the offline diags is that it's hardcoded that Q2 is derived from dQ2 + pQ2 so this would not work currently.